mantle/cmd/kola/switchkernel.go

```diff
@@ -0,0 +1,249 @@
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmdSwitchKernel = &cobra.Command{
+		RunE:    runSwitchKernel,
+		PreRunE: preRun,
+		Use:     "switch-kernel",
+		Short:   "Test on switching between Default and RT Kernel",
+	}
+
+	rtKernelRpmDir string
+)
+
+var (
+	homeDir            = `/var/home/core`
+	switchKernelScript = `#!/usr/bin/env bash
+	# This script is a shameless translation of: https://github.com/openshift/machine-config-operator/blob/f363c7be6d2d506d900e196fa2e2d05ca08b93b6/pkg/daemon/update.go#L651
+	# Usage:
+	# switch-kernel oldkernel newkernel rt-kernel-repo
+	# {oldkernel, newkernel}: either of {default, rt-kernel}
+	# rt-kernel-repo (optional): repository of kernel-rt packages
+	set -xeuo pipefail
+	
+	FROM_KERNEL="${1:?old kernel should be either of \{default, rt-kernel\}}"
+	TO_KERNEL="${2:?new kernel should be either of \{default, rt-kernel\}}"
+	
+	DEFAULT_KERNEL_PKG="kernel kernel-core kernel-modules kernel-modules-extra"
+	RT_KERNEL_PKG="kernel-rt-core kernel-rt-modules kernel-rt-modules-extra"
+	
+	if [[ $FROM_KERNEL == "default" && $TO_KERNEL == "rt-kernel" ]]; then
+		# Switch from default to RT Kernel
+		# https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/update.go#L711
+		RT_KERNEL_REPO=$3
+		if [[ -z $(ls ${RT_KERNEL_REPO}) ]]; then
+			echo "No kernel-rt package available in the repo: ${RT_KERNEL_REPO}"
+			exit 1
+		fi
+	
+		ARGS="override remove ${DEFAULT_KERNEL_PKG}"
+		for RPM in $(ls ${RT_KERNEL_REPO})
+		do
+			ARGS+=" --install ${RT_KERNEL_REPO}/${RPM}"
+		done
+		rpm-ostree ${ARGS}
+	elif [[ $FROM_KERNEL == "rt-kernel" && $TO_KERNEL == "default" ]]; then
+		# Switch from RT Kernel to default
+		# https://github.com/openshift/machine-config-operator/blob/e246be62e7839a086bc4494203472349c406dcae/pkg/daemon/update.go#L667
+		ARGS="override reset ${DEFAULT_KERNEL_PKG}"
+		for PKG in $RT_KERNEL_PKG
+		do
+			ARGS+=" --uninstall $PKG"
+		done
+		rpm-ostree ${ARGS}
+	else
+		echo -e "Invalid options: $@" && exit 1
+	fi
+	`
+)
+
+func init() {
+	cmdSwitchKernel.Flags().StringVar(&rtKernelRpmDir, "kernel-rt", "", "Path to kernel rt rpm directory")
+	cmdSwitchKernel.MarkFlagRequired("kernel-rt")
+	root.AddCommand(cmdSwitchKernel)
+}
+
+func runSwitchKernel(cmd *cobra.Command, args []string) error {
+	// currently only supports RHCOS
+	if kola.Options.Distribution != "rhcos" {
+		return fmt.Errorf("Only supports `rhcos`")
+	}
+
+	var userdata *conf.UserData = conf.Ignition(fmt.Sprintf(`{
+		"ignition": {
+			"version": "2.2.0"
+		},
+		"storage": {
+			"files": [
+				{
+					"filesystem": "root",
+					"path": "/var/home/core/switch-kernel.sh",
+					"contents": {
+						"source": "data:text/plain;base64,%s"
+					},
+					"mode": 110
+				}
+			]
+		}
+	}`, base64.StdEncoding.EncodeToString([]byte(switchKernelScript))))
+
+	flight, err := kola.NewFlight(kolaPlatform)
+	if err != nil {
+		return errors.Wrapf(err, "Flight failed: %v", err)
```

Comment:
The nice thing about `errors.Wrapf` is you *don't* need to do the `: %v", err)`; it does that automatically.  And it also preserves the underlying error so it's not turned into a string.

Reply:
(Search for other places in the code that use it)